fix(package): fix vulnerable deps

antongolub opened the pull request:
Hey, @iarna,

**yargs@8** requires vulnerable version of **mem**. [WS-2018-0236](https://github.com/sindresorhus/mem/commit/da4e4398cb27b602de3bd55f746efa9b4a31702b)
Let's fix it.

closes #2

**UPD 2021-02-14**
* update travis config
  https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration
* require NodeJS >= 10 
  https://github.com/yeoman/update-notifier/releases/tag/v5.0.0
  https://github.com/nodejs/Release#end-of-life-releases

antongolub committed:
fix(package): fix vulnerable deps

homer0 commented:
@antongolub Hi! I got here following a `yarn audit` :P.

You should modify the Travis configuration too, updating those dependencies means dropping support for at least Node 6 (and not sure if 8 too)... and unfortunately, it would be a breaking change release.

antongolub committed:
test: drop nodejs 4, 6 support, add nodejs 12, 14 to test matrix

antongolub commented:
@homer0

I agree, this is reasonable. The latest **yargs** requires nodejs >= 10, but the tests still passed for nodejs v8, so I left it in the test matrix for now.

@iarna,
could you take a look?

iarna commented:
This is why I loathe how security reporting has gone, but unfortunately I don't have a better option.  Correct me if I'm wrong, but this is a DDOS vector? Where an attacker could construct a set of command line arguments that would result in the app leaking memory? Which for a command line tool is a deeply uninteresting sort of problem. Instead, everything gets painted with the brush of long running web apps, who consume data not supplied by the person using the app.

I'm a bit surprised that audit would be pointing at this though -- the updated versions of the transitive dependencies are all compatible -- you should be able to just `npm audit fix` to grab those and leave this module untouched (or use the specific update command from the report). Although... yarn, well, I don't know if it has any facility to individually update a transitive dependency?

Anyway, this PR looks good, I'll bring it in.

homer0 commented:
@iarna Hi!

I'm sorry, like I said, I was following the audit and got to this PR, but after you comment, I re-checked where the issue was, this project `package.json` and, while you are still using `yargs`, you moved it from the prod deps to dev, so it shouldn't trigger any warning (outside this repository).

The path I was following was `semantic-release > @semantic-release/npm > npm > lock-verify > @iarna/cli > yargs > os-locale > mem` and the issue seems to be on `lock-verify`, which requires this package `^1.2.0`, that still used `yargs` as a prod dep.

Now, the weird thing is that when I install the same project with npm, I don't get any vulnerability issue... because while both install `npm@6.14.8`, which requires `lock-verify@^2.1.0`, npm doesn't pick `2.2.1` (which is SemVer compatible), do you have any idea why? my knowledge on this is pretty limited and I alaways assumed that if you have a dependency that has a dependency with a SemVer range, doing a clean install (without lock file) would make the manager resolve on the latest compatible version.

About your question, I was seeing two issues:

- [1084](https://www.npmjs.com/advisories/1084): DDOS triggered because cache is not properly cleaned.
- [1500](https://www.npmjs.com/advisories/1500): A prototype pollution issue from the arguments.

And I agree, this is "too automatic" and doesn't consider context.

Regards!

ffflorian commented:
@iarna

> Anyway, this PR looks good, I'll bring it in.

Any plans when this is going to happen? :slightly_smiling_face:

ffflorian commented:
@iarna This is still open, why? Are there any problems merging this?

antongolub committed:
fix(package): fix vuls

BREAKING CHANGE: require Node.js >= 10.0.0 (https://github.com/nodejs/Release#end-of-life-releases)

antongolub committed:
ci: update travis config
